src/base/utils/version.h

```diff
@@ -31,109 +32,100 @@
 #include <array>
 #include <type_traits>
 
-#include <QDebug>
+#include <QList>
 #include <QString>
+#include <QStringView>
 
-#include "base/exceptions.h"
-#include "base/global.h"
 #include "base/interfaces/istringable.h"
 
 namespace Utils
 {
-    template <typename T, std::size_t N, std::size_t Mandatory = N>
+    // This class provides a default implementation of `isValid()` that should work for most cases
+    // It is ultimately up to the user to decide whether the version numbers are useful/meaningful
+    template <int N, int Mandatory = N>
```

Comment:
>Doesn't std::array use it as size?

std::array is an implementation detail and I don't think it should correlate with it.
Also `std::size_t` is an unsigned number type and I would like to avoid it.